lib/timeout.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+const helper = require('./helper.js');
+const hoek = require('hoek');
+const { waitingJobsPrefix, runningJobsPrefix, queuePrefix } = require('../config/redis');
+const logger = require('screwdriver-logger');
+const DEFAULT_BUILD_TIMEOUT = 60;
+
+/**
+ * Check if the build has timed out
+ * If yes, abort build.
+ * @method check
+ * @return {Promise}
+ */
+async function check(redis) {
+    try {
+        const keys = await redis.hkeys(`${queuePrefix}buildConfigs`);
+
+        if (!keys || keys.length === 0) return;
+
+        await Promise.all(keys.map(async (buildId) => {
+            const json = await redis.hget(
+                `${queuePrefix}buildConfigs`, buildId);
+
+            if (!json) return;
+
+            const buildConfig = JSON.parse(json);
+            const jobId = buildConfig.jobId;
+            const runningKey = `${runningJobsPrefix}${jobId}`;
+            const lastRunningKey = `last_${runningJobsPrefix}${jobId}`;
+            const waitingKey = `${waitingJobsPrefix}${jobId}`;
+            const deleteKey = `deleted_${jobId}_${buildId}`;
+
+            if (!buildConfig) {
```

Comment:
does this check need to be before line 28?

Reply:
Yes, correct. Moving it.